kedro-datasets/kedro_datasets/snowflake/snowpark_dataset.py

```diff
@@ -167,6 +174,7 @@ def __init__(  # noqa: PLR0913
             {"database": self._database, "schema": self._schema}
         )
         self._connection_parameters = connection_parameters
+        self.__session = None
```

Comment:
Are the two underscores intentional?

Reply:
Yes, this is because we use the `_session` property in line 218. Is there a better way to do this?

Reply:
Ahh I see now! Does it need to be a private property? I looked at the other datasets and properties are mostly public.

Reply:
No not necessarily! How would you suggest to change it?

Reply:
If it's okay to make session a public property you'd do:

```python
 @property
    def session(self) -> Session:
        """
        Retrieve or create a session.
        Returns:
            Session: The current session associated with the object.
        """
        if not self._session:
            self._session = self._get_session(self._connection_parameters)
        return self._session
```
And then you can remove the double underscore here as well. But I don't mind too much, so up to you!

Reply:
Will make it in line with the other datasets :)